### PR TITLE
Always log application name/namespace and request object name in logs

### PR DESCRIPTION
**Pull request by MartinBasti**

### CODE REVIEW

File: controllers/binding/snapshotenvironmentbinding_controller.go [hunk 1]
```diff
@@ -75,29 +75,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	application, err := r.getApplicationFromSnapshotEnvironmentBinding(ctx, snapshotEnvironmentBinding)
 	if err != nil {
-		logger.Error(err, "Failed to get Application for ",
-			"SnapshotEnvironmentBinding.Name ", snapshotEnvironmentBinding.Name, "SnapshotEnvironmentBinding.Namespace ", snapshotEnvironmentBinding.Namespace)
+		logger.Error(err, "Failed to get Application from the napshotEnvironmentBinding")
```
Comment (dirgim) [inline controllers/binding/snapshotenvironmentbinding_controller.go:78]: typo: `napshotEnvironmentBinding` -> `snapshotEnvironmentBinding`

**Comment by MartinBasti** [comment]
Log examples:
```
{"level":"info","ts":"2023-05-04T14:53:14Z","logger":"controllers.snapshot","caller":"snapshot/snapshot_adapter.go:272","msg":"The Snapshot won't be released.","snapshot":"default/application-sample-4qkgh","application.namespace":"default","application.name":"application-sample","reasons":"the Snapshot hasn't passed all required integration tests"}

{"level":"info","ts":"2023-05-04T14:53:14Z","logger":"controllers.integrationTestScenario","caller":"scenario/scenario_adapter.go:91","msg":"IntegrationTestScenario has no environment defined","integrationTestScenario":"default/example-pass","application.namespace":"default","application.name":"application-sample"}

{"level":"info","ts":"2023-05-04T14:57:00Z","logger":"controllers.snapshot","caller":"snapshot/snapshot_adapter.go:110","msg":"IntegrationTestscenario pipeline has been created","snapshot":"default/application-sample-cvzln","application.namespace":"default","application.name":"application-sample","namespace":"default","name":"application-sample-cvzln-b82xm","controllerKind":"","audit":"true","action":"ADD","integrationTestScenario.Name":"example-pass"}

{"level":"info","ts":"2023-05-04T14:57:00Z","logger":"controllers.snapshot","caller":"snapshot/snapshot_adapter.go:118","msg":"Snapshot integration status marked as In Progress. Snapshot starts being tested by the integrationPipelineRun","snapshot":"default/application-sample-cvzln","application.namespace":"default","application.name":"application-sample","namespace":"default","name":"application-sample-cvzln","controllerKind":"Snapshot","audit":"true","action":"UPDATE"}

```

File: controllers/binding/snapshotenvironmentbinding_controller.go (hunk 1 above)
Comment (MartinBasti) [reply controllers/binding/snapshotenvironmentbinding_controller.go]: fixed, I like `napshot` though, it sounds like something to put me instantly into a nap

File: helpers/logs.go [hunk 2]
```diff
@@ -63,3 +68,10 @@ func (il *IntegrationLogger) LogAuditEvent(msg string, obj runtime.Object, actio
 
 	log.Info(msg, keysAndValues...)
 }
+
+// WithApp returns a new logger with application.name and application.namespace key-values
+func (il IntegrationLogger) WithApp(app applicationapiv1alpha1.Application) IntegrationLogger {
+	log := il.Logger.WithValues("application.namespace", app.Namespace, "application.name", app.Name)
```
Comment (dirgim) [inline helpers/logs.go:74]: WDYT about shortening to namespaced name, something like this (untested):
```
log := il.Logger.WithValues("application", fmt.Sprintf("%s/%s", application.Namespace, application.Name))
```

Comment (MartinBasti) [reply helpers/logs.go:74]: not sure how hard would be searching in logs for all events in app namespace, but if we always need to search only application namespace/name together, it works for me

Comment (MartinBasti) [reply helpers/logs.go:74]: let's make it shorter, I'll do PR. I cannot really think about usecase where we would need to lookup for something without namespace/name app combination

Comment (MartinBasti) [reply helpers/logs.go]: added

**Review by dirgim — APPROVED** [review]
LGTM

**Review by jsztuka — APPROVED** [review]
lgtm

**Pull request merged**